src/lwc/rd2EntryForm/rd2EntryForm.js

```diff
@@ -0,0 +1,318 @@
+    */
+    handleError(error) {
+        this.errorMessage = this.constrcutErrorMessage(error);
+        this.hasError = true;
+        this.isLoading = false;
+
+        this.template.querySelector("[data-id='submitButton']").disabled = false;
+
+        this.template.querySelector(".slds-modal__header").scrollIntoView();
+    }
+
+    /*******************************************************************************
+    * @description Fires an event to utilDedicatedListener with the cancel action
+    */
+    handleCancel() {
+        fireEvent(this.pageRef, this.listenerEvent, { action: 'cancel', recordId : this.recordId });
+    }
+
+    /*******************************************************************************
+    * @description Fires an event to utilDedicatedListener with the success action
+    */
+    handleSuccess(event) {
+        this.showToast(event.detail.fields.Name.value);
+        fireEvent(this.pageRef, this.listenerEvent, { action: 'success', recordId: event.detail.id});
+    }
+
+    /*******************************************************************************
+    * @description Contruct error wrapper from the error event 
+    *   error.body is the error from apex calls
+    *   error.body.output.errors is for AuraHandledException messages
+    *   error.body.message errors is the error from wired service
+    *   error.detail.output.errors is the error from record-edit-forms
+    */
+   constrcutErrorMessage(error) {
+        let header;
+        let message;
+
+        if (typeof error === 'string' || error instanceof String) {
+            message = error;
+
+        } else if (error.message) {
+            message = error.message;
+
+        } else if ((error.body && error.body.output)) {
+            if (Array.isArray(error.body) &&
+                !error.body.output.errors) {
+                message = error.body.map(e => e.message).join(', ');
+
+            } else if (typeof error.body.message === 'string' &&
+                !error.body.output.errors) {
+                message = error.body.message;
+
+            } else if (error.body.output &&
+                Array.isArray(error.body.output.errors)) {
+                message = error.body.output.errors.map(e => e.message).join(', ');
+            }
+
+        } else if (error.detail && error.detail.output && Array.isArray(error.detail.output.errors)) {
+            header = error.detail.message;
+            message = error.detail.output.errors.map(e => e.message).join(', ');
+
+        } else if (error.body && error.body.message) {
+            message = error.body.message;
+        }
+
+        return {
+            header: header || unknownErrorLabel,
+            detail: message || unknownErrorLabel
+        };
+    }
+
+    /*******************************************************************************
+    * @description Fires a toast event to display success message
+    */
+    showToast(recordName) {
```

Comment:
This function is displaying a toast message on success, so let's either rename it to indicate that or make the method generic and pass arguments that can be either error or success.

Reply:
update to showSuccessToast()